budgetportal/models.py

```diff
@@ -482,17 +483,35 @@ def get_programme_budgets(self):
             return None
         openspending_api = dataset.get_openspending_api()
         financial_year_start = self.get_financial_year().get_starting_year()
-        cuts = [
-            openspending_api.get_financial_year_ref() + ':' + financial_year_start,
-            openspending_api.get_department_name_ref() + ':"' + self.name + '"',
-        ]
+        cuts = []
+
         if self.government.sphere.slug == 'provincial':
-            cuts.append(openspending_api.get_geo_ref() + ':"%s"' % self.government.name)
+            cuts.append(openspending_api.get_geo_ref() +
+                        ':"%s"' % self.government.name)
+
+            cache_name = 'programme-budgets-provincial-{}'.format(
+                self.government.slug)
+        else:
+            cache_name = 'programme-budgets-national'
```

Comment:
I think the caching could happen inside openspending.aggregate using the URL as the cache key - then this module doesn't have to be aware of it. What do you think? We'd obviously need to build the full URL, while we currently pass a params dict to requests.get